packages/common/src/intl/currency.ts

```diff
@@ -0,0 +1,112 @@
+import currency from 'currency.js';
+import { useCurrentLanguage } from './intlHelpers';
+
+const trimCents = (centsString: string) => {
+  const trimmed = String(Number(centsString)).slice(1);
+
+  // If the result is an empty string, return .00
+  if (!trimmed) {
+    return `${centsString[0]}00`;
+  }
+
+  // Otherwise, add a single zero
+  if (trimmed.length === 2) {
+    return `${trimmed}0`;
+  }
+
+  // Other cases, return the full string
+  return trimmed;
+};
+
+/**
+ * This custom formatter is a slight modifiction to the default within
+ * currency.js here: https://github.com/scurker/currency.js/blob/66b7a0c6860d5d30efe8edbf4f8ea016149eab55/src/currency.js#L96-L105
+ *
+ * All it does differently is add the trimming of cents with trimCents.
+ *
+ * Without this, using a high precision i.e. 4, will have a currency formatter to always have
+ * at least 4 decimal digits.
+ *
+ */
+
+export const format: currency.Format = (
+  currency,
+  opts
+  // opts: currency.Options & { groups: RegExp } - this is the correct type.
+) => {
+  if (!currency) return '';
+  const {
+    pattern = '',
+    negativePattern = '',
+    symbol = '$',
+    separator = ',',
+    decimal = '.',
+    // Groups is in the options object, but the type is not right.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    groups = /(\d)(?=(\d{3})+\b)/g,
+  } = opts ?? {};
+
+  // Split the currency string into cents and dollars.
+  const [dollars = '', cents = ''] = ('' + currency)
```

Comment:
thanks currency.js.. the `toString` method returns `0.00` even when the settings say to use `,` as the decimal.
I thought to use `currency.dollar()` and `currency.cents()` to avoid this; but then you have issues with the dropping of trailing zeros that you are testing for.

With the 'fr' language you get this:

<img width="136" alt="Screen Shot 2022-03-03 at 11 50 01 AM" src="https://user-images.githubusercontent.com/9192912/156464259-c1293275-99af-490a-bedb-176bb8a57957.png">

root of the problem is in the `trimCents` which is converting to a number - but the `,` results in `NaN`

I've added a test case for it, have spent a little while fixing but will stop for lunch!

Reply:
Intl is looking better and better 🤔 

Happy to take a look but unsure if you're looking also 😁  let me know!

Reply:
I was.. but I've stopped. if you can pick it up that'd be great

Reply:
Sorry for any time you wasted on this. Testing french was an obvious thing to do.. 🙄 

I've updated the formatter, though I wouldn't describe it as sophisticated, it seems to do the job. I've updated the tests also, which has meant I needed to stop the mock for react-i18n which has had the side effect of causing 100's of tests to fail, which I'll work on next in a new PR..!

Reply:
that's what code reviews are for 😁 
There's a couple of clever changes in the currency formatter; you've deftly sidestepped the problem with `toFixed` by splitting on `.` and then the changes in the `trimCents` is very nice!